lib/cyclonedx/report_python_modules_cyclonedx.rb

```diff
@@ -3,5 +3,19 @@ class ReportPythonModules < Base
     def initialize(scan_report)
       super(scan_report)
     end
+
+    def package_url(dependency)
+      "pkg:#{dependency[:type]}/#{dependency[:name]}#{version_string(dependency, true)}"
+    end
+
+    # Return version string to be used in purl or component
+    def version_string(dependency, is_purl_version = false)
+      # If the dependency is specified in requirements.txt and an absolute version is needed for
+      # the purl return empty
+      return "" if dependency[:dependency_file] == 'requirements.txt' && is_purl_version
```

Comment:
Seems like python doesn't use a lock file but can specify exact versions as show https://iscompatible.readthedocs.io/en/latest/ruby%20encode through here through the `==`. We could aim to include the one's with pinned or semi pinned versions in the purl

Reply:
Yep!